### PR TITLE
Ii 4345 sysdig agent kubernetes

### PR DESCRIPTION
**Pull request by mellv**

### CODE REVIEW

File: reporting_config.json [hunk 1]
```diff
@@ -1 +1 @@
-{"reporting_interval": "1", "prev_endtime": "20180423010646", "keep_file_days": "5", "delta_fields": ["CPU", "DiskRead", "DiskWrite", "NetworkIn", "NetworkOut", "InOctets", "OutOctets", "InErrors", "OutErrors", "InDiscards", "OutDiscards"]}
\ No newline at end of file
+{"keep_file_days": "5", "prev_endtime": "0", "reporting_interval": "1", "delta_fields": ["CPU", "DiskRead", "DiskWrite", "NetworkIn", "NetworkOut", "InOctets", "OutOctets", "InErrors", "OutErrors", "InDiscards", "OutDiscards"]}
```
Comment (ptmr3) [inline reporting_config.json]: why was this changed?

File: sysdig/config.ini [hunk 2]
```diff
@@ -0,0 +1,28 @@
+[insightfinder]
+license_key = a333bd39662ad1edf64e99e97e4b776109827234
```
Comment (ptmr3) [inline sysdig/config.ini]: please add comments in this config to make each item clear and remove your testing values

Comment (mellv) [reply sysdig/config.ini]: Done.

File: reporting_config.json (hunk 1 above)
Comment (mellv) [reply reporting_config.json]: Reverted to master configuration.

File: sysdig/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+```
+./deployment/install.sh -i PROJECT_NAME -u USER_NAME -k LICENSE_KEY -s SAMPLING_INTERVAL_MINUTE -t sysdig -w SERVER_URL
+```
+After using the agent, use command "deactivate" to get out of python virtual environment.
+
+6) In InsightAgent-master directory, make changes to the config file.
+```
+sysdig/config.ini
+```
+Update the api_key. The key can be obtained by going to Settings in user account.
```
Comment (gnegri) [inline sysdig/README.md]: hostname is also required, so you should mention that here.

File: sysdig/getmetrics_sysdig.py [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+        config_parser.read(os.path.abspath(os.path.join(__file__, os.pardir, "config.ini")))
+        try:
+            user_name = config_parser.get('insightfinder', 'user_name')
+            license_key = config_parser.get('insightfinder', 'license_key')
+            project_name = config_parser.get('insightfinder', 'project_name')
+            sampling_interval = config_parser.get('insightfinder', 'sampling_interval')
+            if_http_proxy = config_parser.get('insightfinder', 'if_http_proxy')
+            if_https_proxy = config_parser.get('insightfinder', 'if_https_proxy')
+
+            # host_chunk_size = int(config_parser.get('insightfinder', 'host_chunk_size'))
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: remove if unused

File: sysdig/getmetrics_sysdig.py [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+            # metric_chunk_size = int(config_parser.get('insightfinder', 'metric_chunk_size'))
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: remove if unused

File: sysdig/getmetrics_sysdig.py [hunk 6]
```diff
@@ -0,0 +1,348 @@
...
+            logger.warning(
+                "Agent not correctly configured(project_name). Check config file.")
+            sys.exit(1)
+
+        config_vars = {
+            "userName": user_name,
+            "licenseKey": license_key,
+            "projectName": project_name,
+            "samplingInterval": sampling_interval,
+            # "hostChunkSize": host_chunk_size,
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: remove if unused

File: sysdig/getmetrics_sysdig.py [hunk 7]
```diff
@@ -0,0 +1,348 @@
...
+            # "metricChunkSize": metric_chunk_size,
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: remove if unused

File: sysdig/getmetrics_sysdig.py [hunk 8]
```diff
@@ -0,0 +1,348 @@
...
+
+    return sysdig_config
+
+
+def format_data(res,sub_metric_list):
+    formated_data = []
+
+    for data_dict in res['data']:
+
+        instance = data_dict['d'][0] + '_' + data_dict['d'][1]
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: could either of these values contain a disallowed character?

File: sysdig/getmetrics_sysdig.py [hunk 9]
```diff
@@ -0,0 +1,348 @@
...
+        if len(project_name) == 0:
+            logger.warning(
+                "Agent not correctly configured(project_name). Check config file.")
+            sys.exit(1)
+
+        config_vars = {
+            "userName": user_name,
+            "licenseKey": license_key,
+            "projectName": project_name,
+            "samplingInterval": sampling_interval,
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: should be required or have a default

File: sysdig/getmetrics_sysdig.py [hunk 10]
```diff
@@ -0,0 +1,348 @@
...
+    to_send_data_dict["instanceName"] = socket.gethostname().partition(".")[0]
+    to_send_data_dict["samplingInterval"] = agent_config_vars['samplingInterval']
+    to_send_data_dict["agentType"] = "custom"
+
+    to_send_data_json = json.dumps(to_send_data_dict)
+    print(json.dumps(to_send_data_dict))
+    logger.debug("TotalData: " + str(len(bytearray(to_send_data_json))))
+
+    # send the data
+    post_url = parameters['serverUrl'] + "/customprojectrawdata"
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: use urlparse.urljoin

File: sysdig/getmetrics_sysdig.py [hunk 11]
```diff
@@ -0,0 +1,348 @@
...
+    to_send_data_dict = dict()
+    to_send_data_dict["metricData"] = json.dumps(chunk_metric_data)
+    to_send_data_dict["licenseKey"] = agent_config_vars['licenseKey']
+    to_send_data_dict["projectName"] = agent_config_vars['projectName']
+    to_send_data_dict["userName"] = agent_config_vars['userName']
+    to_send_data_dict["instanceName"] = socket.gethostname().partition(".")[0]
+    to_send_data_dict["samplingInterval"] = agent_config_vars['samplingInterval']
+    to_send_data_dict["agentType"] = "custom"
+
+    to_send_data_json = json.dumps(to_send_data_dict)
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: why do json.dumps here when you do json.loads before sending?

File: sysdig/getmetrics_sysdig.py [hunk 12]
```diff
@@ -0,0 +1,348 @@
...
+            logger.exception(
+                "Too many redirects while flushing to InsightFinder.")
+            break
+        except requests.exceptions.RequestException as e:
+            logger.exception(
+                "Exception while flushing to InsightFinder.")
+            break
+
+    logger.error(
+        "Failed to flush to InsightFinder! Gave up after %d attempts.", ATTEMPTS)
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: replace _ in the for statement and use that iterator variable instead of ATTEMPTS

File: sysdig/getmetrics_sysdig.py [hunk 13]
```diff
@@ -0,0 +1,348 @@
...
+    sdc_token = sysdig_config['sysdig_api_key']
+    hostname = sysdig_config['hostname']
+
+
+    #
+    # Instantiate the SDC client
+    #
+    if len(sysdig_config['httpProxy']) != 0:
+        sdclient = SdcClient(sdc_token, sdc_url='http://app.sysdigcloud.com', ssl_verify=False)
+    if len(sysdig_config['httpsProxy']) != 0:
```
Comment (gnegri) [inline sysdig/getmetrics_sysdig.py]: elif

File: sysdig/getmetrics_sysdig.py (hunk 12 above)
Comment (mellv) [reply sysdig/getmetrics_sysdig.py]: The iterator variable is used only in the format_data() function to off set the dictionary to avoid the first two metrics namely container_id and host.HostName. While the Attempts variable is used to handle retrys. 
The _  was used  in the for loop as the value of ATTEMPTS variable was not used anywhere else.

Comment (gnegri) [reply sysdig/getmetrics_sysdig.py]: I wasn't talking about that loop. I'm talking about the case where the loop breaks early - requests.exceptions.TooManyRedirects or requests.exceptions.RequestException. It could have given up after 1 or 2 tries.

File: sysdig/getmetrics_sysdig.py (hunk 13 above)
Comment (mellv) [reply sysdig/getmetrics_sysdig.py]: Done

File: sysdig/getmetrics_sysdig.py (hunk 11 above)
Comment (mellv) [reply sysdig/getmetrics_sysdig.py]: corrected.

File: sysdig/getmetrics_sysdig.py (hunk 9 above)
Comment (mellv) [reply sysdig/getmetrics_sysdig.py]: It should be required. Made changes to check it its empty.

File: sysdig/config.ini [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
```
Comment (gnegri) [inline sysdig/config.ini]: remove all values in here.
